src/orion/executor/dask_backend.py

```diff
@@ -1,13 +1,52 @@
-from orion.executor.base import BaseExecutor
+import traceback
+from multiprocessing import TimeoutError as PyTimeoutError
+from multiprocessing import Value
+
+from orion.executor.base import AsyncException, AsyncResult, BaseExecutor
 
 try:
-    from dask.distributed import Client, get_client, get_worker, rejoin, secede
+    from dask.distributed import (
+        Client,
+        TimeoutError,
+        get_client,
+        get_worker,
+        rejoin,
+        secede,
+    )
 
     HAS_DASK = True
 except ImportError:
     HAS_DASK = False
 
 
+class _Future:
+    """Wraps a Dask Future"""
+
+    def __init__(self, future):
+        self.future = future
+
+    def get(self, timeout=None):
+        try:
+            return self.future.result(timeout)
+        except TimeoutError as e:
+            raise PyTimeoutError from e
+
+    def wait(self, timeout=None):
+        try:
+            self.future.result(timeout)
+        except TimeoutError:
+            pass
+
+    def ready(self):
+        return self.future.done()
+
+    def successful(self):
```

Comment:
I can't find this method in concurrent.futures.Future. We should have a base class specifying the interface of Future If we depart from python's Future.

Reply:
https://docs.python.org/3/library/multiprocessing.html#multiprocessing.pool.AsyncResult.successful